core/internal/features/features_test.go

```diff
@@ -531,7 +536,9 @@ observationSource   = """
 		assert.Equal(t, []*string(nil), run.Errors)
 
 		testutils.WaitForLogMessage(t, o, "Sending transaction")
-		b.Commit() // Needs at least two confirmations
+		b.Commit()                        // Needs at least two confirmations
+		time.Sleep(10 * time.Millisecond) // wait for finalizer to process confirmed tx
```

Comment:
I don't know where exactly causing the race condition, sleep right after the first committing state is the only solution I found so far

Reply:
Can you elaborate? Sleeping is not a fix for a true data race.

Reply:
Yeah this was the temporarily fix I had. It's been a while since I looked at this, need some time to recall exactly what's wrong.

The issue, from my observation, was a race condition that finalizer and confirmer are both running in parallel processing new head from mailbox. If the confirmer gets runs first, everything is good. However, if finalizer process (`processFinalizedHead`) the new head before confirmer (`ResumePendingTaskRuns`), if will not mark tx to be finalized successfully. Later causing confirmer not able to pickup pending callback txs (needs to be finalized state)